logstash-core/lib/logstash/runner.rb

```diff
@@ -440,6 +442,21 @@ def execute
     @log_fd.close if @log_fd
   end # def self.main
 
+  def running_as_root
+    running_as_root = setting("on_superuser")
+    running_as_root_allowed_options = { "allow" => "ALLOW", "warn" => "WARN", "block" => "BLOCK" }
+    if running_as_root_allowed_options.has_value?(running_as_root) == false
+      logger.warn("Allowed options on on_superuser are WARN, BLOCK and ALLOW. Falling back to default WARN.")
+      running_as_root = running_as_root_allowed_options["warn"]
+    end
```

Comment:
If we set the list of possible values in the setting definition we can do away with this code.

Reply:
Thanks!
I also figured out that if I add `on_superuser ` with options in `environment.rb` declaration, we don't need this lines.
```
Setting::String.new("on_superuser", "WARN", true, ["WARN", "BLOCK", "ALLOW"]),
```